### PR TITLE
Improved legendre polynomial for negative n, added test cases for itself

### LINKED ISSUE #13916: Legendre polynomials should not rewrite "L_{-n}(x)  --->  L_{n-1}(x)"
[Suggested 1st issue].  

The following should return the 3rd degree Legendre polynomial, `5*x**3/2 - 3*x/2`: 
```
from sympy import *; var('x k')
legendre(5 - k, x).subs(k, 2)
```
Instead, I get "ValueError: The index n must be nonnegative integer (got -4)". 

The culprit is a [strange rewriting rule](https://github.com/sympy/sympy/blob/97b7ec4949c4ecb9f6f4db65f4db96ef3b9e53a0/sympy/functions/special/polynomials.py#L789):
```
            # L_{-n}(x)  --->  L_{n-1}(x)
            if n.could_extract_minus_sign():
                return legendre(-n - S.One, x)
```
The function rewrites `legendre(5 - k, x)` as `legendre(k - 6, x)` (why?) and then complains that the value k-6 is negative. 

This rewriting rule should be removed and a test added to prevent it from reappearing.

### PR DESCRIPTION
**Pull request by vivonk**
Fixes #13916 
Instead of giving `ValueError` for negative integer in Legendre polynomial

- Replaced negative number `n` by `-n - 1`
- Used [this](http://functions.wolfram.com/Polynomials/LegendreP/02/) convention for Legendre polynomial of negative number which is 
` L_{n}(x)  = L_{-n -1}(x)` where `n` belongs to negative integer  
- Added test cases for this implementation in test_spec_polynomials.py to pass test in build.

### CODE REVIEW

**Comment by vivonk** [comment]
The above build is giving error on an unintended whitespace, I think
https://github.com/sympy/sympy/blob/87e0000b0655c00f2b00c6f9fb825829c48df53d/sympy/functions/special/tests/test_spec_polynomials.py#L9
@jksuom  @normalhuman please review it and tell if anything is to be improved

**Comment by normalhuman** [comment]
Click details and scroll to the bottom to see what is the reason of error. Specifically: you should remove [this assertion](https://github.com/vivonk/sympy/blob/d5295fda856e7a31e4ba887b0c0765f69dd43fc2/sympy/functions/special/tests/test_spec_polynomials.py#L94) since we no longer raise an error for negative n.

**Comment by normalhuman** [comment]
Also, the words "Fixed #13916" should appear in the first paragraph of the PR description, not in the title. In the title the number is not very helpful because one can't click it. When placed in the first paragraph of the description, it will result in the issue being closed when the PR is merged. See [Writing pull request title and description](https://github.com/sympy/sympy/wiki/Development-workflow#writing-pull-request-title-and-description)

**Comment by vivonk** [comment]
Ok get it, thanks

**Comment by vivonk** [comment]
@normalhuman  Man, it still giving build error on the same tests after removing the ValueError line in test_spec_polynomials.py. Can you help me out with this, mean what this error shows as I am unable to understand the Travis job log. [Here](https://travis-ci.org/sympy/sympy/jobs/328932563#L5693)

File: sympy/functions/special/tests/test_spec_polynomials.py [hunk 1]
```diff
@@ -160,9 +174,6 @@ def test_assoc_legendre():
 
     assert Plm(n, 0, x) == legendre(n, x)
 
-    raises(ValueError, lambda: Plm(-1, 0, x))
-    raises(ValueError, lambda: Plm(0, 1, x))
```
Comment (normalhuman) [inline sympy/functions/special/tests/test_spec_polynomials.py]: Why did you remove the second assertion here? I think it still raises ValueError, because the current implementation of associate Legendre polynomials insists that abs('2nd index') must be <= '1st index'

**Review by normalhuman — CHANGES_REQUESTED** [review]

**Comment by vivonk** [comment]
It's giving error for the above changes mean for the changes in legendre

**Comment by vivonk** [comment]
Ok, correct. Let me change

File: sympy/functions/special/tests/test_spec_polynomials.py (hunk 1 above)
Comment (vivonk) [reply sympy/functions/special/tests/test_spec_polynomials.py]: But the  value error for negative should be removed

Comment (vivonk) [reply sympy/functions/special/tests/test_spec_polynomials.py]: Done 👍

**Review by vivonk — COMMENTED** [review]
Suggested changes done

**Comment by vivonk** [comment]
@normalhuman  it's done :+1: . Now you can review it and tell me about any improvement that I have to do with this.

**Review by normalhuman — APPROVED** [review]

**Comment by vivonk** [comment]
@normalhuman can you merge it, if everything is ok?

**Comment by normalhuman** [comment]
Everything looks ok to me know, but I can't merge in this repo, so your PR will have to wait for someone who can.

**Comment by vivonk** [comment]
@asmeurer  Please, can you check this?

File: sympy/functions/special/tests/test_spec_polynomials.py [hunk 2]
```diff
@@ -107,6 +106,21 @@ def test_legendre():
     assert legendre(10, 0) != 0
     assert legendre(11, 0) == 0
 
+    assert legendre(-1, x) == 1
```
Comment (jksuom) [inline sympy/functions/special/tests/test_spec_polynomials.py]: It is probably not necessary to repeat *all* of the tests for positive `n` with negative `-n - 1`. A few could be chosen.

**Comment by vivonk** [comment]
It's just for show that it work fine as well for negative

**Comment by vivonk** [comment]
@jksuom  should I remove some test cases?

**Comment by jksuom** [comment]
I think that not all of the new tests are useful since they only repeat earlier tests. It would suffice to choose a few and remove the rest.

**Review by vivonk — COMMENTED** [review]
correct

**Comment by vivonk** [comment]
@jksuom can you look at now??

File: sympy/functions/special/tests/test_spec_polynomials.py (hunk 2 above)
Comment (vivonk) [reply sympy/functions/special/tests/test_spec_polynomials.py]: done :+1:

**Comment by vivonk** [comment]
Ping @jksuom  @asmeurer  can you check now?

**Comment by jksuom** [comment]
It seems that the original issue `legendre(5 - k, x).subs(k, 2)` is not tested yet.

**Comment by vivonk** [comment]
@jksuom  I did not add to testing file but tested on local. Here is the clip of it
above test
Should I add this test to the main testing file too? but I don't think it worth as we are more in the normal test cases, not in the test cases of issues.

**Comment by jksuom** [comment]
> Should I add this test to the main testing file too?

I think that should be done, with a reference to the issue.

File: sympy/functions/special/tests/test_spec_polynomials.py [hunk 3]
```diff
@@ -107,6 +106,11 @@ def test_legendre():
...
+    
+    var('k')
```
Comment (jksuom) [inline sympy/functions/special/tests/test_spec_polynomials.py]: It is preferable to use `k = Symbol('k')` in a test file.

**Comment by vivonk** [comment]
@jksuom done, please review it

**Comment by jksuom** [comment]
Looks good, thanks!
